CDTDatastore/query/CDTQIndexManager.m

```diff
@@ -214,8 +214,31 @@ - (instancetype)initUsingDatastore:(CDTDatastore *)datastore error:(NSError *__a
  */
 - (NSString *)ensureIndexed:(NSArray * /* NSString */)fieldNames
 {
-    CDTLogError(CDTQ_LOG_CONTEXT, @"-ensureIndexed: not implemented");
-    return nil;
+    // So we don't break semvar w.r.t CDTQIndex and nullablity, we need to
```

Comment:
Well ideally, we generate the index name on the db queue as we do in sync-android removing the risk of a clashing name. However that would mean the semantics around CDTQIndex and indexName would change, it would become nullable where as before it wouldn't have been. So imo the comment lays out why we are doing the generation here and not there. However that being said, do you think this should be removed or improved upon?